sklearn/preprocessing/data.py

```diff
@@ -1325,6 +1325,21 @@ def normalize(X, norm='l2', axis=1, copy=True, return_norm=False):
     return_norm : boolean, default False
         whether to return the computed norms
 
+    Returns
+    -------
+    X : {array-like, sparse matrix}, shape [n_samples, n_features]
+        Normalized input X.
+
+    norms : array
+        When X is sparse, an NotImplementedError will be raised for
+        norm 'l1' or 'l2'; an array with same shape with X.data,
```

Comment:
Oh I see. Would it be better if it returns a sparse matrix with shape `[row_size/col_size,1]` or a dense 1-d array with shape `[row_size/col_size,]`? I am not sure whether there is a way to generate a sparse 1-d array...